lua/nvim-treesitter.lua

```diff
@@ -18,41 +17,25 @@ function M.setup(lang)
     for _, mod in pairs(configs.available_modules()) do
       if configs.is_enabled(mod, ft) then
         local cmd = string.format("lua require'nvim-treesitter.%s'.attach()", mod)
-        api.nvim_command(string.format("autocmd FileType %s %s", ft, cmd))
+        api.nvim_command(string.format("autocmd NvimTreesitter FileType %s %s", ft, cmd))
       end
     end
+    local cmd = string.format("lua require'nvim-treesitter.state'.attach_to_buffer(%s)", ft)
+    api.nvim_command(string.format('autocmd NvimTreesitter FileType %s %s', ft, cmd))
   end
+
+  state.run_update()
 end
 
 function M.statusline(indicator_size)
   local indicator_size = indicator_size or 1000
-  local expr = require"nvim-treesitter.utils".expression_at_point()
-  local current_node =
-    require'nvim-treesitter.node_movement'.current_node[api.nvim_get_current_buf()]
-
-  local indicator = ""
-  while expr and (#indicator + #(expr:type()) + 5) < indicator_size do
-
-    local prefix = ""
-    if expr:parent() then
-      prefix = "->"
-    end
+  local buf_state = state.get_buf_state()
+  if not buf_state then return "" end
 
+  local current_node = buf_state.current_node
+  if not current_node then return "" end
 
-    if expr == current_node then
-      indicator = string.format("%s[%s]%s", prefix, expr:type(), indicator)
-    else
-      indicator = prefix .. expr:type() .. indicator
-    end
-
-    expr = expr:parent()
-  end
-
-  if expr then
-    return "..." .. indicator
-  else
-    return indicator
-  end
+  return current_node:sexpr()
```

Comment:
Why changing this ? It was useful to see the full path to the current node.

Reply:
Mmhh that's why i asked you to review this. `sexpr` already gives you the full node, i don't really know if getting the whole path will hold in a statusline. But i haven't tried it.

Reply:
The way it worked was displaying the path from root to the current node, truncating if it is too long, like so : 
```
...->block->call_expression->identifier
```
Or if everything fits:
```
source_file->function_item->block
```

(Examples using the Rust parser)